src/extension.ts

```diff
@@ -673,7 +673,18 @@ class ExtensionManager implements vscode.Disposable {
         }
         const drv = await cmt.getCMakeDriverInstance();
         const opts = drv ? drv.expansionOptions : undefined;
-        const env = await effectiveKitEnvironment(kit, opts);
+        let env: Map<string, string> = new Map<string, string>();
+        if (kit) {
+          env = await effectiveKitEnvironment(kit, opts);
+        } else if (cmt.configurePreset) {
```

Comment:
I don't think you should need `else if`, just `else`, unless you plan to validate a second time that kit || preset was used (i.e. in case the `if` on line 663 changes and you want to add a final `else` that asserts or something)

Reply:
I had that if because of red squiggles (typescript didn't figure out that cmt.configurePreset was not null) and I didn't realize that I can use !.